cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/generator/InputInterpreterImpl.java

```diff
@@ -68,23 +69,53 @@ public Object read(String type, Path path, Charset inputCharset, Object... addit
     public Object read(Path path, Charset inputCharset, Object... additionalArguments) throws InputReaderException {
         Set<String> keySet = PluginRegistry.getTriggerInterpreterKeySet();
         // We first try to find an input reader that is most likely readable
+        HashMap<String, Boolean> readableCache = new HashMap<String, Boolean>();
```

Comment:
Please declare maps always as `Map` interface, just the instance could be of `HashMap`

Reply:
adjusted.